Add initial documentation for reusing shared content (#189)
grafana/writers-toolkit: docs/sources/writing-guide/reuse-shared-content.md

> ---

Reviewer: General comment. The writers' toolkit repo uses only _index.md or index.md files for content.

Author: Our latest recommendation in Writers' Toolkit is to prefer pages unless bundles are needed: https://grafana.com/docs/writers-toolkit/writing-guide/documentation-structure/#:~:text=If%20you%20don%E2%80%99t%20know%20whether%20you%20need%20to%20mount%20a%20leaf%20bundle%2C%20you%20probably%20don%E2%80%99t%20and%20default%20to%20using%20pages.

title: Reuse shared content
description: How to reuse chunks of content between pages
keywords:
  - content reuse
  - shared content
---

# Reuse shared content

> Shared content is a sentence, paragraph, or whole page of text that makes sense in multiple contexts.

Reviewer: ```suggestion
Shared content is a sentence, paragraph, or page that makes sense in multiple contexts.
```

> Reusing shared content propagates changes to a single source file to one or more destination pages.

Reviewer 2: ```suggestion
Reusing shared content propagates changes from a single source file to one or more destination pages.
```

This topic describes how to extract and share a chunk of content to multiple pages.

## Before you begin

- Identify a chunk of content that should be reused in multiple pages.

## Steps

To reuse shared content:

1. Identify the sharing project.

>    If the shared content is to be reused in a single project, then the sharing project is that project.

Reviewer: my brain did somersaults trying to parse this. had to read it a couple of times to fully understand it. I recommend adding an example of both use cases, which will help to clarify these words.

Author: Tried rewording in https://github.com/grafana/writers-toolkit/pull/189/commits/addd9f9a389d73653cf6634412f2b0a4fecb8871

   However, if shared content is to be reused across multiple projects, for example Tempo and Grafana Enterprise Traces, then you must choose which project shares the content.

>    Prefer sharing from open source projects.

Reviewer: ```suggestion
   Because we rely on external contributions, sharing from open-source projects is preferred.
```

>    Prefer sharing from open source projects.

Author: You got it spot on, thank you for the suggestion.

1. In the sharing project, create the `docs/sources/shared/` directory if it does not exist.

> 1. In the sharing project, create the file `docs/sources/shared/index.md` with the following contents:

Author: ```suggestion
1. In the sharing project, create the file `docs/sources/shared/index.md` if it does not exist, with the following contents:
```

   ```markdown
   ---
   headless: true
   ---
   ```

   The `index.md` file tells Hugo that the `docs/sources/shared` directory is a leaf bundle which is necessary for the `docs/shared` shortcode to access files stored within.

>    For more information about leaf bundles, refer to [Pages and page bundles]({{< relref "./documentation-structure/#pages-and-page-bundles" >}}).

Reviewer: general comment, it's hard to paint the picture of what these steps look like. I would be inclined to add a screenshot that shows what we want and to add an explanation.

Author: Added in https://github.com/grafana/writers-toolkit/pull/189/commits/51ebb9701a796b2aad50b64e69c8e1fb607bad29

1. In the sharing project, create a file for the shared content in the `docs/sources/shared/` directory.

>    The file should be named to reflect its contents.

Reviewer: ```suggestion
1. Name the file to reflect its contents.
```

>    The file can be in a subdirectory, but that subdirectory must be in the `docs/sources/shared/` directory.

Reviewer: we should insist on subdirectories within shared. it'll help keep the shared content more organized imo. maybe there's a reason not to do that?

Author: I think directory structure is a tough thing to provide general guidance on. It imposes a hierarchy on the contents that might not always be appropriate. I agree that using subdirectories absolutely can help with organization but it can also isolate pages and I am not sure I can explain the guidelines succinctly.

   The `docs/shared` shortcode cannot lookup files outside of this directory.

>    The contents of the file should be the chunk of writing you wish to reuse.

Reviewer: ```suggestion
   The file's contents should be the chunk of writing you want to reuse.
```
